src/App.tsx

```diff
@@ -3,20 +3,45 @@ import {
   ChakraProvider,
   Box,
   Text,
-  Link,
-  VStack,
-  Code,
-  Grid,
   theme,
+  NumberInput,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  Button,
+  Input,
+  Stack,
+  NumberInputField,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper
 } from "@chakra-ui/react"
-import { ColorModeSwitcher } from "./ColorModeSwitcher"
-import { Logo } from "./Logo"
+import {CreateCategoryModal} from "./CreateCategoryModal"
+
+function ModalImplementation() {
+  const [goals, setGoals] = React.useState(["Working out", "Studying", "Reading"]);
+
+  return (
+    <Stack>
+      {goals.map((goal, index) => (
+        <Text key={index}>{goal}</Text>
+      ))}
+      <CreateCategoryModal goals = {goals}/>
+    </Stack>
+  )
+}
 
 export const App = () => (
   <ChakraProvider theme={theme}>
     <Box textAlign="center" fontSize="xl">
       <Text>Time Tracker</Text>
       <Text>Built with ChakraUI</Text>
+      <ModalImplementation/>
```

Comment:
Let's pull from `master` to get the latest changes from that other PR, then instead of adding the modal directly in this `App` component, we probably want to add it to where the sidebar button is for creating a new goal

Reply:
I agree, it was in here purely for demonstration purposes.